### PR TITLE
Part 2 - Improve ValidateCount attribute error message

### PR DESCRIPTION
**Pull request by iSazonov**
Based on discussion https://github.com/PowerShell/PowerShell/pull/3596#discussion_r113464969

Leave only two message for:
1. Count is not in allowed range.
2. Count is not exactly equal allowed value.

/cc @lzybkr @mklement0

### CODE REVIEW

**Comment by iSazonov** [comment]
Mac build temporary failed.

File: src/System.Management.Automation/engine/Attributes.cs [hunk 1]
```diff
@@ -1311,22 +1311,15 @@ protected override void Validate(object arguments, EngineIntrinsics engineIntrin
...
-                    null, Metadata.ValidateCountFailure,
-                    MinLength, MaxLength, len);
-            }
-
-            if (len < MinLength)
-            {
-                throw new ValidationMetadataException("ValidateCountSmallerThanMin",
-                    null, Metadata.ValidateCountFailure,
-                    MinLength, MaxLength, len);
+                throw new ValidationMetadataException("ValidateCountExactlyFailure",
```
Comment (mklement0) [inline src/System.Management.Automation/engine/Attributes.cs]: LGTM overall, but I suggest removing the "ly": `ValidateCountExactlyFailure` -> `ValidateCountExactFailure`

Comment (iSazonov) [inline src/System.Management.Automation/engine/Attributes.cs]: Thanks! Fixed.

**Review by lzybkr — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@lzybkr the FullyQualifiedErrorId changed with this PR, does it need to be reviewed by powershell committee as a breaking change?

**Comment by lzybkr** [comment]
We should document the change, but it's an unlikely breaking change - changing the exception type would be worth discussing, but in this specific case, the error id isn't very interesting.

**Pull request merged**